benchmarks/nemo/nemolite2d/manual_versions/psykal_acc/time_step_mod.f90

```diff
@@ -58,10 +58,29 @@ subroutine invoke_time_step(istp, ssha, ssha_u, ssha_v, &
     hv%data_on_device     = .TRUE.
     ht%data_on_device     = .TRUE.
 
+    ! Specify device data retrieving methods
```

Comment:
Well, that's a design choice..., I think I prefer the verbosity of assigning the function per field here as this will be auto-generated anyway and it allows for more flexibility in the future (e.g. allowing eager or lazy synchonization read_from_device methods attached to different fields), although we probably won't explore this right now.

Reply:
Good points. Flexibility is good and autogeneration is the future :-)